nix/tests.nix

```diff
@@ -8,6 +8,7 @@
 , diffutils
 , git
 , haskell
+, haskellPackages
```

Comment:
@monacoremo This is certainly not the right way to include this here, I assume. Need it to get `hpc-codecov`.

When I started nix-shell the first time I saw some downloads of ghc 8.8.4 or something like that, while everything else is running on 8.8.3. So I assume, this is wrong?

Reply:
Hm, yes... but the setup I chose might be a bit inelegant, and might be better fixed with an overlay that picks the default haskellPackages of our choosing.

To fix this short term: import just `haskell` and pass in our `compiler` variable from default.nix, then use `pkgs.haskell.packages."${compiler}".[...]`

Reply:
So... I just did this in default.nix:

```
  haskellPackages = pkgs.haskell.packages."${compiler}";
...
  tests =
    pkgs.callPackage nix/tests.nix { inherit haskellPackages ...; };
```

It looked like it worked: When I restarted nix-shell, it downloaded / built hpc again.

Does that make sense, remotely?

Reply:
It would be even cleaner I think to request a concrete derivation we want (hpc, hpc-codecov?) at the top of this file instead of the broad haskellPackages, and then to pass it in explicitly in default.nix:

```
tests = ... callPackage ... { inherit ...; hpc = pkgs.haskell....; }

```